target/linux/pistachio/base-files/lib/pistachio.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/sh
+#
+# Copyright (C) 2017 OpenWrt.org
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+PISTACHIO_BOARD_NAME=
+PISTACHIO_MODEL=
+
+pistachio_board_detect() {
+	local name=$(grep "^machine" /proc/cpuinfo | sed "s/machine.*: IMG \(.*\) (\(.*\))/\1/g" | awk '{print tolower($1)}')
+	local model=$(grep "^machine" /proc/cpuinfo | sed "s/machine.*: IMG \(.*\) (\(.*\))/\2/g")
```

Comment:
I think img,pistachio-marduk as a string to identify the board in base-files is fine.
As for IMG Marduk (Creator Ci40), I think that should be changed in the device tree file instead.